### ISSUE TITLE
Support for texture maps

### ISSUE THREAD

**Issue by rcmaniac25** [MosaicManufacturing/canvas-issues#173]
**Is your feature request related to a problem? Please describe.**
For more complex painting/external painting.

**Describe the solution you'd like**
At a basic level: combine stamping with a texture import. If the model doesn't have texture information, then don't allow for importing a texture (grey out the option). If it does, then allow for importing a texture and do a color reduction just like stamping.

At a more advanced level: the texture is a template. In some simulation video games that allow modding, you specify a texture template. This template is only a handful of colors and are hardcoded according to the game. The game then swaps the colors in realtime. Basically: game wants to draw a pixel, picks a point on the texture that corresponds to that texture, it's 0xFF0000 (red). Red texture goes to variable 1, which is currently set to some random color, and that gets assigned to the pixel.

Basically, the advanced mode lets someone specify a texture without needing to worry about filament colors/brands. Then in CANVAS, they can specify the filament to go with that texture template color. This can possibly be baked into the model (so the texture doesn't need to be saved), though this probably requires something like #142 so that smooth lines can exist if baking into the model occurs.

**Additional context**
Idea came to mind while talking with Derek Vogt at ERRF and later online. Later saw a couple model makers would make their models, then paint them digitally... but obviously, we would only get the model and not the painted item. One of them demoed Substance Painter for painting models.
